### ISSUE TITLE
Oracle implementation

### ISSUE THREAD

**Issue by hieronx** [centrifuge/protocol#7]
- Factory to deploy oracle contract, intended to have 1 per pool, although it could be shared
- On deployment, set the `feeder`/`owner`. For now just 1 account.
- When providing a new quote, also include a `quote_reference_time` param. Linear pricing will be based off of this timestamp rather than the `block.timestamp` it was provided.

```solidity
function getIndicativeQuote(..) external view returns (uint256);

// Uses linear pricing
function getIndicativeQuote(..) external view returns (uint256);

function setQuote(..) external onlyFeeder;
```

**Comment by lemunozm** [comment]
> When providing a new quote, also include a quote_reference_time param. Linear pricing will be based off of this timestamp rather than the block.timestamp it was provided.

@hieronx could you extend this point? 🙏🏻. 

I am not sure how to compute the indicative quote. I need at least two feeds to compute the linear pricing. When the user calls `setQuote`/`feed`, it gives me one point of the line. From where do I get the other one? In CentrifugeChain it was the automatically get from notional (for the price value) and maturity (for the reference time). Values that do not have here.

**Comment by hieronx** [comment]
Yeah true, this needs more thinking. I would propose we revisit this once we start speccing the bond item contract.

**Comment by lemunozm** [comment]
Ok! will do in the meantime the simplified version just with `getQuote()`

**Comment by lemunozm** [comment]
Curious about what's the point of creating a factory contract instead of letting the user/feeder just redeploy the contract by themself

**Comment by hieronx** [comment]
@lemunozm three points:
- Contract deployments through a factory are cheaper
- The factory emits events such that an indexer can track all deployments
- You can enable deterministic deployments (same address across all chains) using CREATE2

**Comment by wischli** [comment]
> * When providing a new quote, also include a `quote_reference_time` param. Linear pricing will be based off of this timestamp rather than the `block.timestamp` it was provided

I will apply this to `LinearAccrual` once necessary. Before, I would like to split the contract into a library as discussed last week.

**Comment by lemunozm** [comment]
Note that this "linear" `pricing is another kind of linear than linear-accrual. It's just an interpolated line between two values to get an approximated one.
